src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTestCommon/CollectionRules/Actions/CallbackAction.cs

```diff
@@ -73,6 +75,20 @@ public ICollectionRuleAction Create(IProcessInfo processInfo, BaseRecordOptions
         }
     }
 
+    internal sealed class CallbackActionDescriptor : ICollectionRuleActionDescriptor
```

Comment:
I assume that these cannot be handled by a single generic class because the configuration source generator wouldn't be able to intercept the Bind call? Because if it could, I think all of these could be collapsed into one class.